### PR TITLE
Add new script for using Mock RPC Server

### PR DESCRIPTION
**Pull request by hamstu**
### Description

Small addition to start webpack-dev-server for Publish and have /rpc requests go to a server running on port :9000, which is the mock RPC I have locally. See that here: https://github.com/bufferapp/mock-rpc-server/

### Best Practices Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] I have tested different user stories.
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

### ⚠️ Pre-deploy Checklist

- [ ] I have manually loaded the URL of the staging deployment and confirmed it's working (see below**)
- [ ] I have confirmed that the Cypress E2E test is passing (_Note: we have a known issue where it doesn't run on some PRs. Hopefully this will be fixed soon._)
- [ ] I have confirmed that all other tests are passing.

** To visit the URL of the staging deployment, please vlick the link in the bot comment, or look under checks at the bottom of this PR and click "Details" next to `bufferbotbrains/cicd-buffer-publish`
image

### CODE REVIEW

File: webpack.config.mock.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      '/rpc': {
+        target: 'https://publish.local.buffer.com:9000/',
+        changeOrigin: true,
+        secure: false,
+      },
+      '/pusher/auth': {
+        target: 'https://publish.local.buffer.com/',
+        changeOrigin: true,
+        secure: false,
+      },
```
Comment (hamstu) [inline webpack.config.mock.js:18]: ⚠️ when running this way there isn't anything running to handle the /pusher/auth request (yet), but that can be addressed in a future PR.

Comment (MayaUribe) [reply webpack.config.mock.js:18]: @hamstu should we add a comment here somewhere to reflect that? :)

**Review by MayaUribe — APPROVED** [review]
I just left a question @hamstu but looks great other than that! 🚀🌮

File: webpack.config.mock.js (hunk 1 above)
Comment (hamstu) [reply webpack.config.mock.js:18]: Good one! I think maybe I'll add a note to the mock-rpc-server README on this! Thanks @MayaUribe

**Pull request merged**